improver/utilities/spatial.py

```diff
@@ -99,11 +101,22 @@ def calculate_grid_spacing(
     coord = cube.coord(axis=axis).copy()
     coord.convert_units(units)
     diffs = np.unique(np.diff(coord.points))
+
     if len(diffs) > 1:
-        raise ValueError(
-            "Coordinate {} points are not equally spaced".format(coord.name())
-        )
-    return diffs[0]
+        diffs_diff = np.diff(diffs)
+        diffs_mean = np.mean(diffs)
+
+        if np.any(abs(diffs_diff) > rtol * abs(diffs_mean)):
+            raise ValueError(
+                "Coordinate {} points are not equally spaced".format(coord.name())
+            )
+        # use average spacing for considering tiny difference due to rounding error
+        spacing = (coord.points[-1] - coord.points[0]) / (len(coord.points) - 1.0)
+
+    else:
+        spacing = diffs[0]
```

Comment:
This suggestion makes  neat on code. On the other hand, the user of the function loses the flexibility of specifying rtol. 
Australia domain grid at BOM has some tiny uneven spacing issue. I prefer to have rtol inside arguments  in case we may be required to use it for tuning up rtol slightly.

Reply:
The `np.allclose` method takes `rtol` as an argument, so it can be added when it is found to be needed. I don't want the argument present on this function until something actually uses it. We have regretted putting in functionality into IMPROVER too early in the past, we end up having to maintain something we don't even want.

Reply:
I think using `np.unique` is a bit strange with floats. I'm wondering if it could be made simpler with:

```python
    diffs = np.diff(coord.points)
    diffs_mean = np.mean(diffs)
    if np.any((diffs - diffs_mean) / diffs_mean > rtol):
        raise ValueError(f"Coordinate {coord.name()} points are not equally spaced")
    else:
        return diffs_mean
```

The `if` checks if any "relative error" between a spacing (each `diffs[i]`) and the mean spacing is greater than the relative tolerance. Otherwise just return the mean, which is the same as `spacing = (coord.points[-1] - coord.points[0]) / (len(coord.points) - 1.0)`.

Reply:
From @MoseleyS (which was posted as an edit on my comment above):

> I think using np.unique is a bit strange with floats.

It depends what you are trying to do. Up until now, this function has only ever been used on grids that we control and know to be precisely regular. Therefore, this is fine. @zfan001's regridding work is the first time we've tried to use this on external grids. I'd rather not change the behaviour for internal grids as it helps us be confident that our meta-data are as we intend. A quick test shows that `np.allclose(..., rtol=0.)` returns False for machine-precision errors, so having a default value `rtol=0`. rather than my earlier suggestion (`rtol=None`) would be better.